src/api/app/controllers/webui/request_controller.rb

```diff
@@ -116,8 +116,7 @@ def show
       @refresh = @action[:diff_not_cached]
 
       # Handling build results
-      @building_package = building_package
-      @building_project = @bs_request.staged_request? ? @bs_request.staging_project : @building_package.project
+      @staging_project = @bs_request.staging_project.name unless @bs_request.staging_project_id.nil?
```

Comment:
Why this `@building_project` -> `@staging_project` renaming? This can also happen outside of staging projects, right?

Reply:
Sorry for the confusion, `@staging_project` is not replacing `@building_project`.

I stopped passing `@building_package` and `@building_project` objects to the partial and pass only their names. But the project name can be either the project given by the action or a staging project. That's why I'm retrieving the staging project name in that line and afterwards I pass one or the other:

```
= render partial: 'webui/request/beta_show_tabs/build_results', locals: { project: @staging_project || @action[:sprj],
```

Once inside the partial, how do we know if the passed project name corresponds to the staging project or the project in the action? Before, I passed  `@bs_request` to the partial and checked inside. Now I decided not to pass the `@bs_request` but only a boolean `is_staged_request: @staging_project.present?`.

I hope it's clearer now, please ask me if not.

Reply:
Okay, clear